stage #1/module #6. IO, NIO/lesson #1. IO/content.md

```diff
@@ -0,0 +1,127 @@
+There are many byte stream classes. To demonstrate how byte streams work, we'll focus on the file I/O byte streams, [FileInputStream](https://docs.oracle.com/javase/8/docs/api/java/io/FileInputStream.html) and [FileOutputStream](https://docs.oracle.com/javase/8/docs/api/java/io/FileOutputStream.html) which read and write files. Other kinds of byte streams are used in much the same way; they differ mainly in the way they are constructed.
+
+Following is an example which makes use of these two classes to copy an input file into an output file using byte stream, one byte at a time:
+
+| ![img_2.png](assets/img_2.png) |
+|-------------------------|
+| Byte Stream usage       |
+
+*CopyFile* spends most of its time in a simple loop that reads the input stream and writes the output stream, one byte at a time, as shown in the following figure.
+![img_4.png](assets/img_4.png)
+
+Method “read()” reads one element and return it’s value. When operation read is ended, then returned “-1”. Method “write(c)” writes one element into file.
+You can read about other Byte Streams [here](https://www.javatpoint.com/bytestream-classes-in-java).
+![img_5.png](assets/img_5.png)
+
+## Character Streams
+
+The Java platform stores character values using Unicode conventions. Character stream I/O automatically translates this internal format to and from the local character set. In Western locales, the local character set is usually
+an 8-bit superset of ASCII.
+
+For most applications, I/O with character streams is no more complicated than I/O with byte streams. 
+Input and output done with stream classes automatically translates to and from the local character set. 
+A program that uses character streams in place of byte streams automatically adapts to the local character set and 
+ready for internationalization — all without extra effort by the programmer.
+
+Character streams are often "wrappers" for byte streams. 
+The character stream uses the byte stream to perform the physical I/O, 
+while the character stream handles translation between characters and bytes. *FileReader*, for example,
+uses *FileInputStream*, while *FileWriter* uses *FileOutputStream*. The major difference is that *FileReader* reads two bytes at one time and *FileWriter* writes two bytes at one time. 
+Therefore, previous program could be simply rewritten by replacing *FileOutputStream* on *FileWriter*  and *FileInputStream* on *FileReader*.
+
+![img_6.png](assets/img_6.png)
+
+There are two general-purpose byte-to-character "bridge" streams: [InputStreamReader](https://docs.oracle.com/javase/8/docs/api/java/io/InputStreamReader.html) and [OutputStreamWriter](https://docs.oracle.com/javase/8/docs/api/java/io/InputStreamReader.html). 
```

Comment:
You use link InputStreamReader for OutputStreamWriter. Could you fix it?

Reply:
Fixed